File: 1108_DefanginganIPAddress/DefanginganIPAddress.py

class Solution:
    def defangIPaddr(self, address: str) -> str:
        ##own
        res = ""
        for c in address:
            if(c == '.'):
                res += "[.]"
            else: 
                res+=c
        return res
        ##fun
        # return address.replace(".","[.]")
        ##fun
        # res = address.split(".")
        # return "[.]".join(res)

File: 1108_DefanginganIPAddress/DefanginganIPAddress.java

class Solution {
    public String defangIPaddr(String address) {
        // String res = "";
        // for(char c : address.toCharArray()){
        //     if(c == '.') res += "[.]";
        //     else res+=c;
        // }
        // return res;
        return address.replace(".","[.]");
    }
}
